### ISSUE TITLE
Tax Reporting Bug When Calculated by US County

### ISSUE THREAD

**Issue by jmccall75** [Automattic/woocommerce-services#1620]
Hey, we run a Woo store that uses this plugin to calculate tax rates for a shop based in North Carolina. In NC, the rate varies by the county in which the customer resides, so each NC customer would be paying a slightly different tax rate. This works perfectly **except** that there’s no way to tell from the tax report which county the tax was collected for.

If we go to WooCommerce > Reports > Taxes, and choose Taxes by code, we see a list of six different tax rates for NC that have been collected, ranging from 6.75% to 7.2%. However, in the “Tax” column, each rate has exactly the same code: “US-NC-NC TAX-1”.

Here's a screenshot: https://cld.wthms.co/oXzceX

3 rows of taxes tied to identical tax codes makes the report pretty useless.

**Comment by DanReyLop** [comment]
Thanks for the detailed report, that was helpful to see exactly where the issue was :)

I've created an issue in TaxJar's official plugin: https://github.com/taxjar/taxjar-woocommerce-plugin/issues/109.

When the issue is fixed in their repository, we'll backport it into WooCommerce Services (we don't want WooCommerce Service's TaxJar integration to diverge from the TaxJar plugin). So, you can subscribe to that issue if you want.

The solution seems simple (calling the tax name `CITY_NAME Tax` instead of `STATE_CODE Tax`), but I'm worried that a report with hundreds of rows wouldn't be that useful either.

**Comment by chariotcreative** [comment]
@DanReyLop Any updates on this? We are having the exact same issue with the County not appearing in any tax reports. Even after we manually change the "Tax Name" in Settings --> Taxes --> Standard Rates section. We are hesitant to edit WC core files because they'll just get overridden with the next WC update. @jmccall75 have you gotten this resolved using any other methods?

**Comment by jmccall75** [comment]
No, we didn't. This is a client site and I believe they're figuring out the counties manually by reviewing the actual orders.

**Comment by chariotcreative** [comment]
Ok thanks for your quick response. Same thing here - it's a client site. We are considering using the official TaxJar plugin and having to eat their $17/mo fee for our client because... I guess WooCommerce was not the best recommendation in regards to reports on features they, themselves, offer.

**Comment by DanReyLop** [comment]
@chariotcreative There hasn't been any activity in the issue I created on the TaxJar's plugin repository over 6 months ago: taxjar/taxjar-woocommerce-plugin#109. As I mentioned in my previous comment, the bug must be fixed in the TaxJar plugin before we backport it into WooCommerce Services. You may have better luck commenting in that issue or submitting a PR to the TaxJar plugin repo.

**Comment by mouligreenlaw** [comment]
Another case here: 
3331480-zen

**Comment by tijosh** [comment]
7024417-hc

**Comment by Abdalsalaam** [comment]
Fixed in #2899.
